### DISCUSSION TITLE
Is it possible to wrap plugins with nui.nvim?

### DISCUSSION THREAD

**Discussion by ghost** [MunifTanjim/nui.nvim#329] [Q&A]
Intent: Make floating windows and etc. standardized

Is it possible to wrap mason.nvim, lazy.nvim and etc with nui.nvim? If so, where should I start?

**Comment by MunifTanjim** [comment]
It depends.

If those plugins expose an API to replace the UI, it will be possible... e.g. `telescope.nvim` supports this with the `create_layout` API.

If those plugins use the standard `vim.ui.*` methods, you can override those methods to use `nui.nvim`. Check the wiki to see how that's done.

Otherwise, it will not be possible. You'll need to ask those plugins' maintainers to add support for replacing UI elements.
